pipeline/test/test_processor.py

```diff
@@ -0,0 +1,788 @@
+    #  Leaving these cases for now as the bug is out of the scope of this test.
+    input = [
+        "Targets of incentive measures are all new construction of buildings, or extensions, renovations, improvements, remodel-",
+        "ling, pre-installation/repairs of cooling system equipment that contribute to the improvement of energy conservation perfor",
+    ]
+    expected = [
+        "Targets of incentive measures are all new construction of buildings, or extensions, renovations, improvements, ",
+        "remodelling, pre-installation/repairs of cooling system equipment that contribute to the improvement of energy conservation perfor",
+    ]
+    output = adobe_hyphenation_postprocessor._rewrap_hyphenated_words(input)
+    assert output == expected
+
+    # Test case 2: Keep the hyphenation if the word is a sub-word of the English Language.
+    # cf. pg 1 of cclw-8149-5b598bd3e88c4ce99f44cfbd283c9679.pdf
+    input_2 = [
+        "The extreme demand on these two plants makes the ",
+        "power supply highly unreliable. During the dry season the ",
+        "power supply deficit can increase from 13 to 23.5 MW, ",
+        "meaning that people increasingly turn to expensive, diesel-",
+        "powered generators as an alternative source of energy. ",
+        "Technical and commercial losses of 24.4% have further ",
+        "reduced the availability of electricity. Burundi’s energy envi",
+    ]
+    expected_2 = [
+        "The extreme demand on these two plants makes the ",
+        "power supply highly unreliable. During the dry season the ",
+        "power supply deficit can increase from 13 to 23.5 MW, ",
+        "meaning that people increasingly turn to expensive, ",
+        "diesel-powered generators as an alternative source of energy. ",
+        "Technical and commercial losses of 24.4% have further ",
+        "reduced the availability of electricity. Burundi’s energy envi",
+    ]
+    output_2 = adobe_hyphenation_postprocessor._rewrap_hyphenated_words(input_2)
+
+    assert output_2 == expected_2
+
+
+def test_adobe_list_find_first_occurrence(adobe_list_postprocessor):
+    regex = adobe_list_postprocessor.regex_pattern
+    # Case 1:
+    input = ["Document", "L[3]", "LI", "LBody", "Span[2]"]
+    expected_output = "L[3]"
+    actual_output = adobe_list_postprocessor._find_first_occurrence(regex, input)
+    assert actual_output == expected_output
+    # Case 2:
+    input_2 = ["Document", "L", "LI", "LBody", "Span[2]"]
+    expected_output_2 = "L"
+    actual_output_2 = adobe_list_postprocessor._find_first_occurrence(regex, input_2)
+    assert actual_output_2 == expected_output_2
+    # Case 3: No matching regex.
+    input_3 = ["Document", "P"]
+    actual_output_3 = adobe_list_postprocessor._find_first_occurrence(regex, input_3)
+    assert not actual_output_3
+
+
+def test_adobe_text_styling_merge_test_blocks(adobe_text_styling_postprocessor):
```

Comment:
If you visit the actual document from the test case, it shows that the logic isn't particularly useful in this particular case. As above, I chose this case to further illustrate the combinatorial possibility of cases to deal with in the Adobe API. Nevertheless, tested functionality passes.

Reply:
this is great!